File: lib/domain/stroop_stimulus.dart
import 'package:stroop_overload/core/stroop_color.dart';

class StroopStimulus {
  const StroopStimulus({
    required this.wordLabel,
    required this.inkColor,
    this.audioColor,
    this.bgDistractor,
  });

  final StroopColor wordLabel;
  final StroopColor inkColor;
  final StroopColor? audioColor;
  final StroopColor? bgDistractor;

  StroopColor get correctAnswer => inkColor;
  bool get isIncongruent => wordLabel != inkColor;
}

File: app/src/main/kotlin/com/stroopoverload/domain/StroopStimulus.kt
package com.stroopoverload.domain

import com.stroopoverload.core.StroopColor

data class StroopStimulus(
    val wordLabel: StroopColor,
    val inkColor: StroopColor,
    val audioColor: StroopColor? = null,
    val bgDistractor: StroopColor? = null,
) {
    val correctAnswer: StroopColor get() = inkColor
    val isIncongruent: Boolean get() = wordLabel != inkColor
}
